src/main/java/com/google/sps/servlets/GroceryServlet.java

```diff
@@ -27,39 +27,108 @@
+                response.sendRedirect("/grocery.html");
+                return;
+            }
+        }
+    }
+    
     // adds item to datastore
     String grocery = request.getParameter("groceryItem");
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Entity groceryEntity = new Entity("Grocery");
     if( grocery != null && !grocery.equals("")) {
         groceryEntity.setProperty("grocery", grocery);
+        groceryEntity.setProperty(FAMILY_ID, familyID);
         datastore.put(groceryEntity);
-    }
-               
-    doGet(request,response);  
+    }     
     response.sendRedirect("/grocery.html");
   }
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      // creates arraylist and starts query
-      ArrayList<String> groceryList = new ArrayList<String>();
-      Query query = new Query("Grocery");
-      PreparedQuery results = datastore.prepare(query);
-        
-      //loads entities into arraylist to be printed
-      for (Entity entity : results.asIterable()) {
-          String groceryItem = (String) entity.getProperty("grocery");
-        groceryList.add(groceryItem);
-      }
-      Gson gson = new Gson();
-      response.setContentType("application/json;");
-      response.getWriter().println(gson.toJson(groceryList));
+
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
+
+    // checks if user is a part of a family and returns an error if they aren't
+    String userEmail = userService.getCurrentUser().getEmail();
+    Query query = new Query("UserInfo")
+        .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, userEmail));
+    PreparedQuery results = datastore.prepare(query);
+    Entity userInfoEntity = results.asSingleEntity();
+    if (userInfoEntity == null) {
+        response.setContentType("text/html;");
+        response.getWriter().println("You do not belong to a family yet!");
+    return;
+    }
+
+    // creates arraylist and starts query
+    ArrayList<String> groceryList = new ArrayList<String>();
+    long familyID = (long) userInfoEntity.getProperty(FAMILY_ID);
+    Query groceryQuery = new Query(GROCERY)
+      .setFilter(new Query.FilterPredicate(FAMILY_ID, Query.FilterOperator.EQUAL, familyID));   
+    PreparedQuery familyGrocery= datastore.prepare(groceryQuery);
+    groceryList = checkGroceries(familyGrocery);   
+
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(groceryList));
     }
+    // returns items from the query that match the users familyID
+  private ArrayList<String> checkGroceries(PreparedQuery familyGrocery) {
```

Comment:
how about fetchGroceriesForFamily as a name?

Also, I would suggest putting lines 113-115 inside the method. That way it encapsulates all the fetching groceries for a family ID logic.